### PR TITLE
Implemented Uniforms derive macro

### PR DESCRIPTION
**Pull request by nikarh**
Hi,

As the title says, this PR contains a derive macro for `Uniforms` trait, #2

### CODE REVIEW

**Comment by nikarh** [comment]
Hey, don't want to be annoying, but is it possible to get somebody to look at this PR?

File: src/lib.rs [hunk 1]
```diff
@@ -150,3 +159,88 @@ fn impl_glium_vertex_derive(ast: &DeriveInput) -> TokenStream {
...
+                },
+                NestedMeta::Meta(Meta::Path(ref path)) => {
+                    let ident = path.get_ident().expect("Expected ident got path");
+                    panic!("Unknown field attribute {}", ident);
+                },
+                _ => (),
+            }
+        };
+
+        quote_spanned! {field.span()=>
```
Comment (est31) [inline src/lib.rs:194]: Can you add a space before the = ?

**Review by est31 — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (nikarh) [reply src/lib.rs:194]: Oh, that's the syntax of `quote_spanned`, there should be no space

https://docs.rs/proc-quote/latest/proc_quote/macro.quote_spanned.html

File: src/lib.rs [hunk 2]
```diff
@@ -150,3 +159,88 @@ fn impl_glium_vertex_derive(ast: &DeriveInput) -> TokenStream {
...
+                    let ident = path.get_ident().expect("Expected ident got path");
+
+                    if ident == "attr" {
+                        uniform_name = quote!(#lit);
+                    } else {
+                        panic!("Unknown field attribute {}", ident);
+                    }
+                },
+                NestedMeta::Meta(Meta::Path(ref path)) => {
+                    let ident = path.get_ident().expect("Expected ident got path");
```
Comment (est31) [inline src/lib.rs]: This could use some punctuation

**Review by est31 — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 3]
```diff
@@ -150,3 +159,88 @@ fn impl_glium_vertex_derive(ast: &DeriveInput) -> TokenStream {
...
+                        let ident = path.get_ident().expect("Expected ident got path");
+
+                        if ident == "glium" {
+                            nested.iter().cloned().collect()
+                        } else {
+                            Vec::new()
+                        }
+                    },
+                    _ => {
+                        Vec::new()
```
Comment (est31) [inline src/lib.rs]: Maybe you could put this Vec::new() at the end of the closure and then drop both the else and compact this arm to just {}? The collect call would get a return at the front.

**Comment by nikarh** [comment]
Fixed both comments + extracted field attribute parser as a function, because it's literally ctrl+c ctrl+v from the other macro

**Review by est31 — APPROVED** [review]
Thank you!

**Pull request merged**